### PR TITLE
improve pr ci workflow

### PR DESCRIPTION
**Pull request by dfrankland**
* remove redundant jobs
* test before build for better dev cycle
* beef up x86_64 linux machines for faster test and build

### CODE REVIEW

File: .github/workflows/pull_request.yml [hunk 1]
```diff
@@ -79,52 +79,6 @@ jobs:
...
-    strategy:
-      matrix:
-        include:
-          - target: x86_64-unknown-linux-musl
-            runs-on: public-amd64-small
-          - target: aarch64-unknown-linux-musl
-            runs-on: public-arm64-medium
-    name: Unit Tests for ${{ matrix.target }}
-    runs-on: ${{ matrix.runs-on }}
-    steps:
```
Comment (dfrankland) [inline .github/workflows/pull_request.yml:119]: This job was redundant with the release build job. Both build and test

File: .github/workflows/pull_request.yml [hunk 2]
```diff
@@ -40,16 +40,16 @@ jobs:
...
-
       - name: Install latest nextest release
         uses: taiki-e/install-action@nextest
 
       - name: Run tests
         run: cargo nextest run --profile=ci
 
+      - name: Build ${{ matrix.target }} target
+        if: "!cancelled()"
+        run: cargo build -q --all --release --target ${{ matrix.target }}
```
Comment (dfrankland) [inline .github/workflows/pull_request.yml:51]: builds take 5+ minutes, so testing before building makes way more sense

File: .github/workflows/pull_request.yml [hunk 3]
```diff
@@ -64,7 +64,7 @@ jobs:
       matrix:
         include:
           - target: x86_64-unknown-linux-musl
-            runs-on: public-amd64-small
+            runs-on: public-amd64-2xlarge
```
Comment (dfrankland) [inline .github/workflows/pull_request.yml:67]: using a larger machine helps speed up tests and builds quite a lot. I would love if we could have more chunky mac and ARM linux machines

**Comment by dfrankland** [comment]
@pat-trunk-io (@gnalh ?) I don't have permission to change the required checks for this repo. Can we remove the requirements for "Unit Tests*" jobs? The "Build*" jobs already test

**Comment by gnalh** [comment]
> @pat-trunk-io (@gnalh ?) I don't have permission to change the required checks for this repo. Can we remove the requirements for "Unit Tests*" jobs? The "Build*" jobs already test

I don't have them either.

**Comment by dfrankland** [comment]
@EliSchleifer perms for this repo?

**Comment by pat-trunk-io** [comment]
Done

**Review by riya-n — APPROVED** [review]

**Pull request merged**